cmd/acadock-monitoring/isgraceful/isgraceful.go

```diff
@@ -0,0 +1,225 @@
+package isgraceful
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"testing"
```

Comment:
question: It seems weird to me to have the `testing` package within a go file which is not for testing purpose.
Are you sure of what you made here?

Reply:
The isgraceful/isgraceful.go is designed for testing.

It uses the same techniques employed in go-utils/graceful, but written in a re-useable way.

I wanted to discuss this with you. I think it would be worth having this moved to go-utils package, so it can be reused to test the graceful behaviour of our other services too (some do not have tests e.g. git-repository-core)

Reply:
isgraceful removed

Reply:
> I wanted to discuss this with you. I think it would be worth having this moved to go-utils package, so it can be reused to test the graceful behaviour of our other services too (some do not have tests e.g. git-repository-core)

Did you talk about this to the team during my days-off? I think it would be great to discuss it with the team like you did for the signal capture :wink:

Reply:
I did not discuss it with the team, but I did do so with Leo.

It has been moved to go-utils/graceful package, but is not-exported. Leo was :+1: to replace the exiting test functions with it. In theory it could be exported, but there is no need for it at the moment.

We should add a ticket for the future to add a function to go-utils/graceful to ListenTCP. There is another SL service that uses tableflip to listenTCP. It would be good if it could use go-utils/graceful library instead.

This could be a ninja task.

Reply:
Really nice! :clap: 
If you want, you can create the issue related to this in the https://github.com/Scalingo/go-utils/issues to not forget about it.
And indeed, it will be a ninja task!